stable/spark/templates/spark-master-deployment.yaml

```diff
@@ -0,0 +1,134 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: default-pyspark
+  namespace: default
+subjects:
+- kind: User
+  name: system:serviceaccount:default:default
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
+  name: edit
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Values.master.name }}
+  labels:
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    component: "{{ .Release.Name }}-{{ .Values.master.component }}"
+spec:
+  type: {{ .Values.master.serviceType }}
+  ports:
+    - port: {{ .Values.master.servicePort }}
+      name: master
+      targetPort: {{ .Values.master.containerPort }}
+    - port: 8888 
+      nodePort: {{ .Values.master.jupyter.nodePort }}
+      name: jupyter
+      targetPort: 8888
+  externalIPs:
+    - "{{ .Values.externalIP.ip }}"
+  selector:
+    component: "{{ .Release.Name }}-{{ .Values.master.component }}"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Values.webUi.name }}
+  labels:
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    component: "{{ .Release.Name }}-{{ .Values.master.component }}"
+spec:
+  type: {{ .Values.webUi.serviceType }}
+  ports:
+    - port: {{ .Values.webUi.servicePort }}
+      nodePort: {{ .Values.webUi.nodePort }}
+      targetPort: {{ .Values.webUi.containerPort }}
+  externalIPs:
```

Comment:
add externalIPs only if specified in values. Take a look at [here](https://github.com/DODAS-TS/helm_charts/blob/master/stable/cachingondemand/templates/cache-redir-svc.yml#L26-L28) for reference